srai/embedders/geovex/embedder.py

```diff
@@ -259,3 +261,80 @@ def _assert_feature_length(self, target_features: list[str], conv_layer_size: in
             raise ValueError(
                 f"The convolutional layers in GeoVex expect >= {conv_layer_size} features."
             )
+
+    def save(self, path: Union[str, Any]) -> None:
+        """
+        Save the model to a directory.
+
+        Args:
+            path (Union[str, Any]): Path to the directory.
+        """
+        # embedder_config must match the constructor signature:
+        # target_features: Union[list[str], OsmTagsFilter, GroupedOsmTagsFilter],
+        # batch_size: Optional[int] = 32,
+        # neighbourhood_radius: int = 4,
+        # convolutional_layers: int = 2,
+        # embedding_size: int = 32,
+        # convolutional_layer_size: int = 256,
+        embedder_config = {
+            "target_features": self.expected_output_features.to_json(),
+            "batch_size": self._batch_size,
+            "neighbourhood_radius": self._r,
+            "convolutional_layers": self._convolutional_layers,
+            "embedding_size": self._embedding_size,
+            "convolutional_layer_size": self._convolutional_layer_size,
+        }
+        self._save(path, embedder_config)
+
+    def _save(self, path: Union[str, Any], embedder_config: dict[str, Any]) -> None:
+        if isinstance(path, str):
+            path = Path(path)
+
+        self._check_is_fitted()
+
+        path.mkdir(parents=True, exist_ok=True)
+
+        # save model and config
+        self._model.save(path / "model.pt")  # type: ignore
+        # combine model config and embedder config
+        if self._model is not None:
+            model_config = self._model.get_config()
```

Comment:
Why is there a check for a model not being `None`? If `self._model` was `None`, the `save` function would throw an exception. 

Should this be a check for a model config, or should the `save` function be moved inside this if statement?

Reply:
I can see in the `Hex2Vec` implementation that there is no such check with `if` statement.